### ISSUE TITLE
Make etc/ci/upload_nightly.sh responsible for putting datetimes into filenames

### ISSUE THREAD

**Issue by aneeshusa** [servo/servo#12128]
Part of #11980.

Currently, our Linux and Mac (although not Android) `./mach package` builders create files whose name include the date and time of building, which is a source of non-determinism. However, we do want to include the date and time for our nightly builds, so that multiple nightlies can coexist in a folder (e.g. `$HOME/downloads`) or S3 bucket (e.g. `servo-builds`). Two parts to this:
- Make `./mach package` output files that always use the same name (e.g. servo.tar.gz). File: `python/servo/package_commands.py`.
- Make `etc/ci/upload_nightly.sh`  insert the appropriate date and time when uploading the built nightly to S3. This depends on #12004 landing.

Bonus points (but not required for the E-easy): add a test that changing the date/time (e.g. with a date/time faking library) does not affect the filename or contents of the output.

**Comment by aneeshusa** [comment]
Note that this needs to be fixed at most a day after Android nightlies go live so that we don't lose old nightlies (OK to lose nightlies before we officially release them IMO).

**Comment by imjacobclark** [comment]
Happy to pick this one up also.

**Comment by imjacobclark** [comment]
To clarify, we no longer wish `mach package` to put date and times on files, rather this be deferred to `etc/ci/upload_nightly.sh` instead? 

@aneeshusa

**Comment by aneeshusa** [comment]
Yes, `./mach package` should not add a date or time; `etc/ci/upload_nightly.sh` should include a date and time in the location to which it uploads.

**Comment by aneeshusa** [comment]
The test I mentioned will essentially require us to package (twice!), so don't worry about that for now; we need to decide about #12103 first.

**Comment by imjacobclark** [comment]
Resolved in: https://github.com/servo/servo/pull/12163

**Comment by KiChjang** [comment]
This is not yet resolved until #12163 is merged to the master branch.
